src/pages/settings/Settings.tsx

```diff
@@ -20,25 +20,30 @@ import ConfigFieldDescription from "pages/settings/ConfigFieldDescription";
 import { toConfigFields } from "util/config";
 import CustomLayout from "components/CustomLayout";
 import PageHeader from "components/PageHeader";
+import { useSupportedFeatures } from "context/useSupportedFeatures";
 
 const Settings: FC = () => {
   const docBaseLink = useDocs();
   const [query, setQuery] = useState("");
   const notify = useNotify();
+  const {
+    hasMetadataConfiguration,
+    settings,
+    isSettingsLoading,
+    settingsError,
+  } = useSupportedFeatures();
 
   const { data: configOptions, isLoading: isConfigOptionsLoading } = useQuery({
-    queryKey: [queryKeys.configOptions],
+    queryKey: [queryKeys.configOptions, hasMetadataConfiguration],
     queryFn: fetchConfigOptions,
```

Comment:
```suggestion
    queryKey: [queryKeys.configOptions],
    queryFn: () => fetchConfigOptions(hasMetadataConfiguration),
```
I doubt the `queryKey` is meant to pass information to  the `queryFn`. It hides the dependency quite well. Better pass the flag directly?

Reply:
Passing directly has also the advantage, that you can't forget to pass the flag.

Reply:
Aaah indeed this is a better approach, will change that :+1: